openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSPatientServiceImpl.java

```diff
@@ -245,6 +250,42 @@ private Patient updatePatient(Config config, Patient patient) {
         return updatedPatient;
     }
 
+    private Patient updatePatientIdentifiers(Config config, Patient patient) {
+        Validate.notNull(patient, "Patient cannot be null");
+        Validate.notEmpty(patient.getUuid(), "Patient Id may not be empty");
+
+        List<Identifier> identifiersList = patientResource.getPatientIdentifierList(config, patient.getUuid());
+        Patient updatedPatient;
+        try {
+            //Patient Identifiers have to be update separately.
+            updatePatientIdentifiers(config, patient, identifiersList);
+            updatedPatient = getPatientByUuid(config, patient.getUuid());
+            eventRelay.sendEventMessage(new MotechEvent(EventKeys.UPDATED_PATIENT_SUBJECT, EventHelper.patientParameters(updatedPatient)));
+        } catch (HttpClientErrorException e) {
+            throw new OpenMRSException("Failed to update OpenMRS patient with id: " + patient.getUuid(), e);
+        }
+        return updatedPatient;
+    }
+
+    private void updatePatientIdentifiers(Config config, Patient patient, List<Identifier> fetchedIdentifierList) {
+        String fetchedIdentifierDisplay;
+        String newIdentifierName;
+
+        for (Identifier fetchedIdentifier : fetchedIdentifierList) {
+            for (Identifier newIdentifier : patient.getIdentifiers()) {
+                //Display field in indetifierType on the OpenMRS server contains the name. Name field there is empty.
+                fetchedIdentifierDisplay = fetchedIdentifier.getIdentifierType().getDisplay();
```

Comment:
Try to add 'v=full' to the request maybe then you will have the name of identifier